src/server/dfly_main.cc

```diff
@@ -323,13 +324,28 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
   auto port = GetFlag(FLAGS_port);
   auto mc_port = GetFlag(FLAGS_memcache_port);
   string unix_sock = GetFlag(FLAGS_unixsocket);
+  mode_t unix_socket_perm;
+  string perm_str = GetFlag(FLAGS_unixsocketperm);
+  if (perm_str.empty()) {
+    // get umask of running process.
+    mode_t umask_val = umask(0);
+    umask(umask_val);
+    unix_socket_perm = 0777 & ~umask_val;
+  } else {
+    if (!absl::numbers_internal::safe_strtoi_base(perm_str, &unix_socket_perm, 8) ||
+        unix_socket_perm > 0777) {
+      LOG(ERROR) << "Invalid unixsocketperm: " << perm_str;
+      exit(1);
```

Comment:
You could use either `QCHECK()` or `LOG(QFATAL)` to achieve an exit with no stack frames or noisy output.
It will also save the `if` clause.

Reply:
we are using google logging library which does not have the QFATAL/QCHECK. From what I see this is only supported in absl logging

Reply:
Oh :facepalm:
I didn't realize absl logging and google logging are different, and that they have different features. That's disappointing :(